File: CohesiveRP.Storage/DataAccessLayer/Settings/Summary/SummarySettings.cs
using System.Text.Json.Serialization;

namespace CohesiveRP.Storage.DataAccessLayer.Settings.Summary
{
    public class SummarySettings
    {
        [JsonPropertyName("shortConfig")]
        public ShortSummaryConfig Short { get; set; } = new();

        [JsonPropertyName("mediumConfig")]
        public ExtensibleSummaryConfig Medium { get; set; } = new();

        [JsonPropertyName("longConfig")]
        public ExtensibleSummaryConfig Long { get; set; } = new();

        [JsonPropertyName("extraConfig")]
        public ExtensibleSummaryConfig Extra { get; set; } = new();

        [JsonPropertyName("overflowConfig")]
        public OverflowSummaryConfig Overflow { get; set; } = new();

        [JsonPropertyName("nbRawMessagesToKeepInContext")]
        public int NbRawMessagesToKeepInContext { get; set; } = 3;
    }
}

File: CohesiveRP.UI.WebFrontend/src/ResponsesDto/settings/BusinessObjects/SummarySettings.ts
export interface ShortSummaryConfig {
  nbMessageInChunk: number;
  maxShortTermSummaryTokens: number;
}

export interface ExtensibleSummaryConfig {
  summarizeLastXTokens: number;
  maxTotalSummariesTokens: number;
}

export interface OverflowSummaryConfig {
  summarizeLastXTokens: number;
  maxOverflowSummaryTokens: number;
}

export interface SummarySettings {
  shortConfig: ShortSummaryConfig;
  mediumConfig: ExtensibleSummaryConfig;
  longConfig: ExtensibleSummaryConfig;
  extraConfig: ExtensibleSummaryConfig;
  overflowConfig: OverflowSummaryConfig;
  nbRawMessagesToKeepInContext: number;
  hotMessagesAmountLimit: number;
  coldMessagesAmountLimit: number;
}
